src/parser.rs

```diff
@@ -152,6 +152,16 @@ impl Library {
         let get_type = elem.attr_required("get-type")?;
         let version = self.read_version(parser, ns_id, elem)?;
         let deprecated_version = self.read_deprecated_version(parser, ns_id, elem)?;
+        let is_fundamental = elem.attr("fundamental").map(|x| x == "1").unwrap_or(false);
+        let (ref_fn, unref_fn) = if is_fundamental {
+            (
+                elem.attr("ref-func").map(ToOwned::to_owned),
+                elem.attr("unref-func").map(ToOwned::to_owned),
+            )
+        } else {
+            (None, None)
+        };
+
```

Comment:
There's also `get-value-func` / `set-value-func` which you can use to implement the `glib::Value` traits automatically here

Reply:
The thing I noticed for those is only GtkExpression & GskRenderNode that have a `get-value-func` / `set-value-func` and for GskRenderNode they are behind a feature guard (were only added in v4_6),  not sure if it's worth the change in gir for now

Reply:
Ideally they would also be added for `GdkEvent` so that the way how they're currently passed through signals in C is less of a disaster but OK with me to do that later. Please create an issue about that then :)

Reply:
Done https://github.com/gtk-rs/gir/issues/1299